### DISCUSSION TITLE
External connectivity for debugging / standalone operation

### DISCUSSION THREAD

**Discussion by kaolpr** [sinara-hw/EEM_FMC_Carrier#12] [Specification]
I see there is no external connectivity that can be used in a standalone mode. Maybe we could add at least a simple USB-UART? Or even SATA connector in an assembly variant (to be used for ethernet connection).

**Comment by gkasprow** [comment]
USB-UART is the simplest form of diagnostics. If we don't have the panel space, the USB connector can be mounted on board (vertical one)

**Reply by marmeladapk** [reply]
For UART I have 1 free pin right now (CFG_PUDC_B). If we want another pin we could either add another GPIO extender, merge P3V3_FMC_ENABLE and VADJ_ENABLE or remove one of the LEDs (we would have only 2 LEDs left then).

**Reply by marmeladapk** [reply]
Ok, I found another pin that is pretty easy to escape. I'll put CP2105 and USB B connector. I feel like vertical micro USB would be too prone to breaking.

**Reply by gkasprow** [reply]
can you fit USB B connector? Maybe use mini-USB

**Reply by kaolpr** [reply]
Maybe there is some THT version of micro-USB? It's just that USB B (and miini-USB) is not that common and I believe looking for such cable in a lab may be a bit cumbersome...

But if it's the only option it's better than nothing ;-).

**Reply by kaolpr** [reply]
Guess I got a bit spoiled with Kasli and new AFC, but I'd rather have at least FT2232 with JTAG and UART on this "service port".

**Reply by gkasprow** [reply]
one must choose micro-USB with 4 THT mounting tabs. There are also true THT versions. I think I saw them in CERN library

**Reply by gkasprow** [reply]
sth like this [https://pl.farnell.com/gct-global-connector-technology/usb3145-30-1-a/micro-usb-2-0-type-b-receptacle/dp/2443141?gclid=CjwKCAiAv_KMBhAzEiwAs-rX1OfwPfDnUbJYxDsGoEJLH0nJvfNx_eEfSikTliDhco5lQoIXyylewBoC9W8QAvD_BwE&mckv=s7zbNbPBx_dc|pcrid|560370199583|plid||kword||match||slid||product|2443141|pgrid|130862485178|ptaid|pla-331900302425|&CMP=KNC-GPL-GEN-SHOPPING-Whoop-10-Nov-2021&gross_price=true](url)

**Reply by marmeladapk** [reply]
I saw those connectors, but they seem to rely on pad support for perpendicular forces. Given that the usb cable probably will be yanking on that connector I feel that sooner or later something will break. USB B connector at least is not more prone to yanking in any direction. Also, is it really a niche cable? And does it matter since it's going to be used only for uart debugging?

**Reply by gkasprow** [reply]
But didn't we decide to go for the pin header?

**Reply by marmeladapk** [reply]
This comment was made before my discussion with kaolpr.

**Comment by kaolpr** [comment]
I was thinking of vertical USB connector too. I mentioned SATA specifically for standalone operation. As this board has relatively low-end FPGA and is not very complex it could be considered a simple standalone FMC carrier as a secondary use-case.

**Reply by gkasprow** [reply]
SATA could be an interesting option. One can add our simple SATA to SFP adapter and use it stand-alone with just 12V supply

**Reply by marmeladapk** [reply]
I'm worried about signal integrity - MGT3 signals already have multiple vias and a switch on their way. +they're routed only on bottom layers.

**Comment by marmeladapk** [comment]
Since using SATA either way requires additional board (SATA-SFP adapter) I have another idea:

For standalone mode we could design an add-in board which would connect to the backplane connector - the board could have:
   - 12V connector (minimum variant)
   - SFP connector connected to MGT (and 12V->3,3V converter) (minimum variant)
   - if needed, uart could be done through LVDS lines (but that would require active components on the board) (medium variant)
   - potentially we could also do usb-jtag + uart + i2c, but that would require ftdi chip which is hard to come by these days (max variant)
   
Nice thing about this idea is that it's also inherently compatible with DIOT_FMC_Carrier. Heck, we could even do PoE in this form factor if someone wanted that.

**Reply by gkasprow** [reply]
makes sense, but if you remove the DIOT part, there won't be a way to connect the debugger..

**Reply by kaolpr** [reply]
But we can use DIOT part only for SFP. Use-case is for standalone operation. It does not have to be two-way convertible. Just keep some sort of "service port" on the main board.

**Comment by marmeladapk** [comment]
After brief discussion with @kaolpr we arrived at the following solution:

- 4 pin UART header (2,54mm) with GND and P3V3 for debug purposes
- Add-in board as described [here](https://github.com/sinara-hw/EEM_FMC_Carrier/discussions/12#discussioncomment-1688181) in max variant - standalone operation requires diot variant

**Reply by jordens** [reply]
Good!

**Reply by gkasprow** [reply]
make sure that you follow pinout from other projects (stabilizer, Thermostat)
